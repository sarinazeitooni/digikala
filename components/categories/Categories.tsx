import React, {useState} from 'react';
import style from './scss/categories.module.scss';
import CategoriesCard from "./categories-card/CategoriesCard";
interface props{
    items : []
}
const Categories = ({items} : props)=>{
    const [data, setData] = useState(items.data);
    return(
        <div className={style['categories-container']}>
            <h3>{items.categoriesTitle}</h3>
            <div className={style['items-container']}>
                {
                    data.map((item)=>{
                        return(
                            <CategoriesCard icon={item.icon} title={item.title} product={item.product}/>
                        )
                    })
                }
            </div>
        </div>
    )
}
export default Categories;