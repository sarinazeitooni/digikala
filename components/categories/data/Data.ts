const Data = {
    categoriesTitle : 'بیش از ۲،۰۰۰،۰۰۰ کالا در دسته‌بندی‌های مختلف',
    data : [
        {
            icon : 'ComputerOutlinedIcon',
            title : 'کالای دیجیتال',
            product : '715000 کالا'
        },
        {
            icon : 'BuildOutlinedIcon',
            title : 'خودرو، ابزار و تجهیزات صنعتی',
            product : '134000 کالا'
        },
        {
            icon : 'GrainOutlinedIcon',
            title : 'مد و پوشاک',
            product : '۷۱۵۰۰۰ کالا'
        },
        {
            icon : 'ChildCareOutlinedIcon',
            title : 'اسباب بازی، کودک و نوزاد',
            product : '64000 کالا'
        },
        {
            icon : 'ShoppingBasketIcon',
            title : 'کالاهای سوپرمارکتی',
            product : '57000 کالا'
        },
        {
            icon : 'FavoriteOutlinedIcon',
            title : 'زیبایی و سلامت',
            product : '92000 کالا'
        },
        {
            icon : 'HomeOutlinedIcon',
            title : 'خانه و آشپزخانه',
            product : '433000 کالا'
        },
        {
            icon : 'BorderColorOutlinedIcon',
            title : 'کتاب، لوازم تحریر و هنر',
            product : '242000 کالا'
        },
        {
            icon : 'ChangeHistoryOutlinedIcon',
            title : 'ورزش و سفر',
            product : '34000 کالا'
        },
        {
            icon : 'GrainOutlinedIcon',
            title : 'محصولات بومی و محلی',
            product : '0 کالا'
        }
    ]
};
 export default Data;