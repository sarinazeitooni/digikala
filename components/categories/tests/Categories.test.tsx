import {shallow} from "enzyme";
import Categories from "../Categories";
import Data from "../data/Data";
describe("categories component" , ()=>{
    it("should render correctly" , ()=>{
        const container = shallow(<Categories items={Data}/>);
        expect(container).toMatchSnapshot();
    })
});