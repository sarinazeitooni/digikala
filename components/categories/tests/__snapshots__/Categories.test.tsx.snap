// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`categories component should render correctly 1`] = `ShallowWrapper {}`;
